miio/pet_water_dispenser.py

```diff
@@ -0,0 +1,227 @@
+import enum
+import logging
+from typing import Any, Dict, List
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .exceptions import DeviceException
+from .miot_device import DeviceStatus, MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_MMGG_PET_WATERER_S1 = "mmgg.pet_waterer.s1"
+MODEL_MMGG_PET_WATERER_S4 = "mmgg.pet_waterer.s4"
+
+SUPPORTED_MODELS: List[str] = [MODEL_MMGG_PET_WATERER_S1]
+
+_MAPPING: Dict[str, Dict[str, int]] = {
+    # https://home.miot-spec.com/spec/mmgg.pet_waterer.s1
+    # https://home.miot-spec.com/spec/mmgg.pet_waterer.s4
+    "cotton_left_time": {"siid": 5, "piid": 1},
+    "reset_cotton_life": {"siid": 5, "aiid": 1},
+    "reset_clean_time": {"siid": 6, "aiid": 1},
+    "fault": {"siid": 2, "piid": 1},
+    "filter_left_time": {"siid": 3, "piid": 1},
+    "indicator_light": {"siid": 4, "piid": 1},
+    "lid_up_flag": {"siid": 7, "piid": 4},  # missing on mmgg.pet_waterer.s4
+    "location": {"siid": 9, "piid": 2},
+    "mode": {"siid": 2, "piid": 3},
+    "no_water_flag": {"siid": 7, "piid": 1},
+    "no_water_time": {"siid": 7, "piid": 2},
+    "on": {"siid": 2, "piid": 2},
+    "pump_block_flag": {"siid": 7, "piid": 3},
+    "remain_clean_time": {"siid": 6, "piid": 1},
+    "reset_filter_life": {"siid": 3, "aiid": 1},
+    "reset_device": {"siid": 8, "aiid": 1},
+    "timezone": {"siid": 9, "piid": 1},
+}
+
+
+class OperatingMode(enum.Enum):
+    common = 1
+    smart = 2
+
+
+class PetWaterDispenserException(DeviceException):
+    pass
+
+
+class PetWaterDispenserStatus(DeviceStatus):
+    """Container for status reports from the Pet Water Dispenser."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def sponge_filter_left_days(self) -> int:
+        """Filter life time remaining in days."""
+        return self.data["filter_left_time"]
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is on."""
+        return self.data["on"]
+
+    @property
+    def mode(self) -> str:
+        """OperatingMode."""
+        return OperatingMode(self.data["mode"]).name
```

Comment:
```suggestion
        return OperatingMode(self.data["mode"])
```
Better just return the mode here